django/contrib/postgres/operations.py

```diff
@@ -261,3 +262,54 @@ def describe(self):
     @property
     def migration_name_fragment(self):
         return 'remove_collation_%s' % self.name.lower()
+
+
+class AddConstraintNotValid(AddConstraint):
+    """
+    Add a table constraint without enforcing validation on existing rows
+    using PostgreSQL's ADD table_constraint NOT VALID syntax (supports only
+    check constraints.)
+    """
+
+    def describe(self):
+        return 'Create constraint %s on model %s without enforcing validation on existing rows' % (
+            self.constraint.name, self.model_name
+        )
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        model = from_state.apps.get_model(app_label, self.model_name)
+        if self.allow_migrate_model(schema_editor.connection.alias, model):
+            constraint_sql = self.constraint.create_sql(model, schema_editor)
+            schema_editor.execute(str(constraint_sql) + " " + "NOT VALID")
+
```

Comment:
Might be worth adjusting `migration_name_fragment` here as well to include a `not_valid` suffix.